src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -32,10 +33,9 @@ public sealed class EmitOptions : IEquatable<EmitOptions>
         public bool IncludePrivateMembers { get; private set; }
 
         /// <summary>
-        /// If not empty then compiler should enable instrumentation.
-        /// The string specifies instrumentation settings.
+        /// Type of instrumentation that should be added to the output binary.
         /// </summary>
-        public string Instrument { get; private set; }
+        public HashSet<string> Instrument { get; private set; }
```

Comment:
A flags enum is limited to 32 (or 64) values. An immutable array of non-flags enum values probably works.